extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -785,13 +794,65 @@ export class AmpA4A extends AMP.BaseElement {
             this.getSafeframePath()
           );
         }
-        // Note: Resolving a .then inside a .then because we need to capture
-        // two fields of fetchResponse, one of which is, itself, a promise,
-        // and one of which isn't.  If we just return
-        // fetchResponse.arrayBuffer(), the next step in the chain will
-        // resolve it to a concrete value, but we'll lose track of
-        // fetchResponse.headers.
-        return fetchResponse.arrayBuffer().then(bytes => {
+
+        const size = this.extractSize(fetchResponse.headers);
```

Comment:
Wanted to call out that I moved this up a step in the promise chain. Seems OK but let me know if anyone thinks this could be problematic.